aten/src/ATen/core/dispatch/Dispatcher.h

```diff
@@ -402,9 +402,70 @@ class TypedOperatorHandle<Return (Args...)> final : public OperatorHandle {
 };
 
 namespace detail {
-template<class... Args> inline void unused_arg_(const Args&...) {}
+template <class... Args> inline void unused_arg_(const Args&...) {}
+
+// ReturnValue is intended to capture return values from Dispatcher unboxed
+// kernel calls.
+// A record function may request to get outputs from the kernel calls. For
+// boxed kernels, it's straightforward, the returned values are in the stack
+// object. The stack can be passed to record functions. For unboxed kernels, we
+// need to handle different kinds of return values, cache them temporarily, then
+// release the values for the actual function call return.
+template <typename ReturnType>
+struct ReturnValue {
+  template <typename F, typename... Args>
+  ReturnValue(
+      F&& kernel,
```

Comment:
if you're not forwarding the kernel anywhere, then `const F& kernel` works and is easier to understand than `F&& kernel`. Same for the other specialization below